lib/model/folder_sendrecv.go

```diff
@@ -582,7 +589,7 @@ func (f *sendReceiveFolder) handleDir(file protocol.FileInfo, snap *db.Snapshot,
 	// that don't result in a conflict.
 	case err == nil && !info.IsDir():
 		// Check that it is what we have in the database.
-		curFile, hasCurFile := f.model.CurrentFolderFile(f.folderID, file.Name)
```

Comment:
This (passing around snap) seems unrelated. An oversight from long ago?

Reply:
That's intentional `model.CurrentFolderFile` now returns an error, which I don't want to have to handle here, because the folder might be missing. Plus it needlessly creates a new snapshot (i.e. yes, the change would also make sense without this PR)..